### DISCUSSION TITLE
Teleport connect source

### DISCUSSION THREAD

**Discussion by Robert904** [gravitational/teleport#37081] [Q&A] [answered]
Hi!
Sorry for the (potentially) stupid question, but I can't find the source code of the "Teleport Connect" client.
Is it me missing something obvious, or maybe it's not open-sourced ?

Many thanks

**Comment by zmb3** [comment, marked as answer]
Most of the source is here: https://github.com/gravitational/teleport/tree/master/web/packages/teleterm

You won't be able to build it from source as the open source build is currently broken due to some private enterprise dependencies. We hope to resolve that next quarter.

**Comment by Robert904** [comment]
Thank you for the answer.
But then, what is the license of the binary version distributed through the website if it contains these private entreprise dependencies ?
I can't find any information on the website, in the app itself or in the code, I only find the licenses for the electron and the chromium components...
Many thanks !

**Reply by klizhentas** [reply]
https://goteleport.com/blog/teleport-oss-switches-to-agpl-v3/#what-is-dual-licensing-and-how-exactly-do-you-dual-license

You can find the answers in the blog post above. We dual license our code and binaries, and the OSS binary (not code), is licensed under Apache 2. The code in OSS repo available to OSS community is licensed with AGPL v3. If any OSS users checks out the repo and compiles it, they would get a binary compiled and licensed under AGPL v3.
